packages/apps/spaces/app/organizations/[id]/components/Timeline/events/logEntry/LogEntryDatePicker.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useRef } from 'react';
+import { Text } from '@ui/typography/Text';
+import { Flex } from '@ui/layout/Flex';
+import { LogEntryWithAliases } from '@organization/components/Timeline/types';
+import { DateTimeUtils } from '@spaces/utils/date';
+import { Box } from '@ui/layout/Box';
+import { DatePicker as ReactDatePicker } from 'react-date-picker';
+import { useUpdateLogEntryMutation } from '@organization/graphql/updateLogEntry.generated';
+import { getGraphQLClient } from '@shared/util/getGraphQLClient';
+import { FormInput } from '@ui/form/Input';
+import {
+  LogEntryUpdateFormDto,
+  LogEntryUpdateFormDtoI,
+} from './LogEntryUpdateFormDto';
+import { useField, useForm } from 'react-inverted-form';
+import { useQueryClient } from '@tanstack/react-query';
+
+export const LogEntryDatePicker: React.FC<{ event: LogEntryWithAliases }> = ({
+  event,
+}) => {
+  const timeoutRef = useRef<NodeJS.Timeout | null>(null);
+
+  const client = getGraphQLClient();
+  const queryClient = useQueryClient();
+
+  const logEntryStartedAtValues = new LogEntryUpdateFormDto(event);
+  const { state } = useForm<LogEntryUpdateFormDtoI>({
+    formId: 'log-entry-update',
+    defaultValues: logEntryStartedAtValues,
+
+    stateReducer: (_, _a, next) => {
+      return next;
+    },
+  });
+  const { getInputProps } = useField('date', 'log-entry-update');
```

Comment:
It's ok for now, but we need to pay attention when we abstract this component to be reused in several places. name and formId will need to be passed via props.

Reply:
this component is not meant to be reused - hence it's in logEntry directory and context and it's explicitly called "LogEntryDatePicker'